File: frontend/src/types/catalogo.ts
// Espejo de backend/app/schemas/catalogo.py (Regla 5)
import type { CategoriaPublic } from "./categoria";
import type { ProductoClientePublic } from "./producto";

export interface CategoriasResponse {
  items: CategoriaPublic[];
}

export interface CatalogoResponse {
  items: ProductoClientePublic[];
  page: number;
  page_size: number;
  total: number;
}

File: backend/app/schemas/catalogo.py
# -*- coding: utf-8 -*-
"""Schemas del catálogo de cliente (ARQ-T2)."""
from .categoria import CategoriaPublic
from .common import SchemaBase
from .producto import ProductoClientePublic


class CategoriasResponse(SchemaBase):
    items: list[CategoriaPublic]


class CatalogoResponse(SchemaBase):
    items: list[ProductoClientePublic]
    page: int
    page_size: int
    total: int
